### ISSUE TITLE
Slow when including lots of blade files

### ISSUE THREAD

**Issue by fgilio** [itsgoingd/clockwork#280]
Hello!
We have this weird issue with requests being painfully slow if we have both Clockwork enabled and the view has lots of Blade `@include` statements. A particular view which does a ton of includes often takes from 15s to 30s to load.

Is there anything related to Blade debugging or something else we can disable to speed this up?
Most views do load in a reasonable time. And that particularly slow view also loads pretty fast when `CLOCKWORK_ENABLE=false`

Thanks!

**Comment by fgilio** [comment]
We were not using the latest version! The project was on 3.0. I've just upgraded to 3.0.1 and the problems seems to be 100% solved :)

I find it a little weird that even while having `viewsData` in the `filters` config the requests were so slow. I guess that default filter is not the only change in 3.0.1, right?

**Comment by itsgoingd** [comment]
Glad that fixed the issue! I've had the same problem with my own projects after updating to 3.0. :)

The filter is implemented on the storage level, so *all* data is always collected, but the filtered data is thrown away before saving, saving disk space but not memory or cpu time.

This is obviously not a very smart implementation, it was good enough in the first Clockwork release, but didn't age well with all the added features.

The `databaseQueries`, `cache` and `events` filter options already had a special handling where we disabled collecting the particular data if they were included. A similar exception was added for `viewData` in 3.0.1.

An overall better filter implementation is already on the todo list for the next major release.
